std/datetime.d

```diff
@@ -1075,6 +1075,42 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        static if (is(size_t == ulong))
+        {
+            return _stdTime;
+        }
+        else
+        {
+            enum uint m = 0x5bd1e995;
+            enum uint n = 16;
+            enum uint r = 24;
+            uint h = n;
+            long k = _stdTime;
+
+            k *= m;
+            k ^= k >> r;
+            k *= m;
+
+            h *= m;
+            h ^= k;
+
+            return h;
+        }
+    }
+
+    @safe unittest
+    {
+        assert(SysTime(DateTime.init, UTC()).toHash == SysTime(0, UTC()).toHash);
+        assert(SysTime(DateTime.init, UTC()).toHash == SysTime(0).toHash);
+        assert(SysTime(Date.init, UTC()).toHash == SysTime(0).toHash);
+        assert(SysTime(0).toHash == SysTime(0).toHash);
```

Comment:
No offense, but these tests aren't very good. They're _all_ testing `SysTime(0)` except with `LocalTime` in some cases instead of `UTC`. There's no point in using `DateTime` or `Date` in these tests, since that's just using different constructors. You might as well just use specific integers. Also range of values should be tested - not just `0`, and you should test at least some values that are not supposed to be equal. It's not required for the hashing function to function as a hashing function (e.g. just returning `42` would do that), but it can be used to test that the specific hashing function was done correctly.